src/smspark/bootstrapper.py

```diff
@@ -63,28 +67,42 @@ def bootstrap_history_server(self) -> None:
 
     def copy_aws_jars(self) -> None:
         self.logger.info("copying aws jars")
-        jar_dest = Bootstrapper.SPARK_PATH + "/jars"
         for f in glob.glob("/usr/share/aws/aws-java-sdk/*.jar"):
-            shutil.copyfile(f, os.path.join(jar_dest, os.path.basename(f)))
-        hadoop_aws_jar = "hadoop-aws-2.8.5-amzn-6.jar"
-        jets3t_jar = "jets3t-0.9.0.jar"
-        shutil.copyfile(
-            os.path.join(Bootstrapper.HADOOP_PATH, hadoop_aws_jar), os.path.join(jar_dest, hadoop_aws_jar),
-        )
-        # this jar required for using s3a client
+            shutil.copyfile(f, os.path.join(self.JAR_DEST, os.path.basename(f)))
+        hadoop_aws_jar = self._get_hadoop_jar()
         shutil.copyfile(
-            os.path.join(Bootstrapper.HADOOP_PATH + "/lib", jets3t_jar), os.path.join(jar_dest, jets3t_jar),
+            os.path.join(Bootstrapper.HADOOP_PATH, hadoop_aws_jar), os.path.join(self.JAR_DEST, hadoop_aws_jar)
         )
+
+        self._copy_optional_jars()
         # copy hmclient (glue data catalog hive metastore client) jars to classpath:
         # https://github.com/awslabs/aws-glue-data-catalog-client-for-apache-hive-metastore
         for f in glob.glob("/usr/share/aws/hmclient/lib/*.jar"):
-            shutil.copyfile(f, os.path.join(jar_dest, os.path.basename(f)))
+            shutil.copyfile(f, os.path.join(self.JAR_DEST, os.path.basename(f)))
+
+    def _get_hadoop_jar(self) -> str:
+        for file_name in os.listdir(Bootstrapper.HADOOP_PATH):
+            if file_name.startswith("hadoop-aws") and file_name.endswith(".jar"):
+                self.logger.info(f"Found hadoop jar {file_name}")
+                return file_name
+
+        raise AlgorithmError("Error finding hadoop jar", caused_by=FileNotFoundError())
```

Comment:
This shouldn't ever happen -- is there a case where it does?

Also: this is a strong reason why we should actually be doing this JAR copying at build time, in the `Dockerfile` rather than at runtime. (This is optional for now).

Reply:
Yes this shouldn't even happen. We should never reach to this line. It's just for completeness.